examples/src/test/kotlin/UpdatesBuildersTest.kt

```diff
@@ -0,0 +1,259 @@
+
+    companion object {
+        val dotenv = dotenv()
+        val client = MongoClient.create(dotenv["MONGODB_CONNECTION_URI"])
+        val database = client.getDatabase("store")
+        val collection = database.getCollection<PaintOrder>("paint_orders")
+
+        @AfterAll
+        @JvmStatic
+        fun afterAll() {
+            runBlocking {
+                database.drop()
+                client.close()
+            }
+        }
+    }
+    // utility function to get the document
+    private suspend fun getDocument() = collection.find(Filters.eq("_id", 1)).first()
+
+    @BeforeEach
+    fun beforeEach() {
+        runBlocking {
+            val date = Date(946756228) // Jan 1, 2000
+            val redPaint = PaintOrder(1, "red", 5, listOf("A", "D", "M"), date)
+            collection.insertOne(redPaint)
+        }
+    }
+
+    @AfterEach
+    fun afterEach() {
+        runBlocking {
+            collection.drop()
+        }
+    }
+
+    @Test
+    fun setUpdateTest() = runBlocking {
+        // :snippet-start: set-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.set(PaintOrder::qty.name, 11)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(11, getDocument().qty)
+    }
+
+    @Test
+    fun unsetUpdateTest() = runBlocking {
+        // :snippet-start: unset-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.unset(PaintOrder::qty.name)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(null, getDocument().qty)
+    }
+
+    @Test
+    fun setOnInsertUpdateTest() = runBlocking {
+        collection.deleteOne(Filters.eq("_id", 1))
+        // :snippet-start: set-on-insert-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.combine(
+            Updates.setOnInsert(PaintOrder::qty.name, 7),
+            Updates.setOnInsert(PaintOrder::color.name, "red"),
+        )
+        collection.updateOne(filter, update, UpdateOptions().upsert(true))
+        // :snippet-end:
+        assertEquals(7, getDocument().qty)
+    }
+
+    @Test
+    fun incUpdateTest() = runBlocking {
+        // :snippet-start: inc-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.inc(PaintOrder::qty.name, 3)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(8, getDocument().qty)
+    }
+
+    @Test
+    fun mulUpdateTest() = runBlocking {
+         // :snippet-start: mul-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.mul(PaintOrder::qty.name, 2)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(10, getDocument().qty)
+    }
+
+    @Test
+    fun renameUpdateTest() = runBlocking {
+        // :snippet-start: rename-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.rename(PaintOrder::qty.name, "quantity")
```

Comment:
This is one place where using a data class doesn't make a lot of sense, as once you rename the field it will no longer be usable by this data class.  

I don't know though whether it's worth worrying about